### DISCUSSION TITLE
Shelly Plus Mini devices

### DISCUSSION THREAD

**Discussion by Th0maz** [arendst/Tasmota#19516] [Ideas]
Hey,
is anybody planing to investigate the new Shelly Plus Mini devices?
I bought a Shelly Plus PM Mini last week and was wondering, if they could be flashed with Tasmota. 
Calibration might be a thing with the PM version.
Thanks
Thomas

EDIT by Tasmota crew: Please follow [this](https://github.com/tasmota/mgos32-to-tasmota32) exactly to convert the device successfully!!

**Comment by sfromis** [comment]
Very much **in testing**, and there is already
https://github.com/tasmota/mgos32-to-tasmota32/releases/tag/C3_Shelly
https://github.com/tasmota/autoconf/tree/main/esp32c3
Template for your Shelly Plus PM Mini should basically be the same, just without the switch/relay pins.

**Reply by sfromis** [reply]
Now there are autoconf setups for all 3.

**Reply by Th0maz** [reply]
Do you know, where I can find valid OTA links to put into the Shelly's firmware URL field?
The "mini" versions seem to be not available, yet.

**Comment by arendst** [comment]
Try http://ota.tasmota.com/tasmota32/shelly/mgos32-to-tasmota32-PlusPMMini.zip

**Reply by Th0maz** [reply]
Tried that before. It now seems to have magically appeared somehow ;-)

**Comment by Th0maz** [comment]
![grafik](https://github.com/arendst/Tasmota/assets/12065628/1c084e76-ef22-40f6-8902-5f53c76767c0)
Seems to be working good 😎👍

**Reply by filipq** [reply]
Hello 

I have tried to flash my Shelly Plus Mini devices today with Tasmota and it worked OK apart from energy readings. 
![ts1](https://github.com/arendst/Tasmota/assets/9423402/ef85e4dd-8aec-46dd-a6a2-d5e502672bd9)
Could you send your Template parameters? 
Mine are as follows.
![ts2](https://github.com/arendst/Tasmota/assets/9423402/3b6cf865-410e-4291-934e-b3d3bb062cc5)
Any help is much appreciated. 
Thanks

**Reply by arendst** [reply]
Update to latest dev release. Latest dev branch only supports BL0942 at 9600 bps used on shelly minis.

Once at v13.1.0.3 make sure GPIO7 is BL0942 RX 2

**Reply by Th0maz** [reply]
@filipq Follow the steps Theo wrote.
Then it should look like this:
![grafik](https://github.com/arendst/Tasmota/assets/12065628/71a5bd1f-fec9-4475-9abb-d4e75eb4eae4)

**Reply by Th0maz** [reply]
And I'm not sure if you need to set `setoption21 1` to use energy monitoring when power is off (since we have no power/relay here).
I set that during testing, so I don't really know if it's necessary.

**Reply by filipq** [reply]
Thank you all for your help. All working fine now. After updating to the latest dev release and changing GPIO7 to BL0942 RX 2 everything working as expected. 

![ts3](https://github.com/arendst/Tasmota/assets/9423402/4f41b0a4-a5c3-451d-8d70-5306b49da86c)

I did not set  setoption21 1.

**Reply by Th0maz** [reply]
> I did not set setoption21 1.

I double-checked with my device.
When `setoption21 ` is set to `0`, I don't see any energy monitoring data.

**Reply by arendst** [reply]
As designed.  The shelly plus pm mini needs so21 1 to report anything. Usually a relay is present which is the basis for reporting.  I think the best solution is updating auto config for the shelly plus pm mini with so21 1 (@Jason2866 )

**Reply by Jason2866** [reply]
added `so21 1` to Autoconfig for the pmmini

**Reply by just-grizzle** [reply]
OT: Just came across this while trying to set up BL0942 on a Kincony M1. Changing BL0942 RX 1 to RX 2 did the magic. So thx about that.
But can anyone give me a hint why this was neccessary?

**Comment by Jason2866** [comment]
Final updated Shelly converter online which supports the new C3 Shelly minis.
The updated [Shelly to Tasmota convert ](https://github.com/tasmota/mgos32-to-tasmota32)process does take care of the new more secured Shelly firmware.
With the new Shelly firmware there was a risk to end in a bricked device with the former version.
This problem is fixed. Currently we have had 100% successful converts in our tests.
Let’s see how long we win the cat and mice game!

**Comment by ghostseven** [comment]
I am unfortunately stuck on 12.5.0 after flashing the linked http://ota.tasmota.com/tasmota32/shelly/mgos32-to-tasmota32-PlusPMMini.zip I cannot get it to update.  Any tips or advice?

If it helps I get Upload Failed

Program flash size is larger than real flash size

When trying to manually upload (not OTA)

I have just tried manually flashing 12.5.0 and that works fine, I just can't go above that.

EDIT: My fault I see I have bricked it, no worries I need to follow the guide more clearly.

**Reply by Jason2866** [reply]
The update fails since the new Tasmota v13.1 does not fit in the Shelly OTA partition slots.
The Autoconfig and safeboot conversion was not done right after the first start of Tasmota.
Without the Autoconfig step, the locked Shelly bootloader is not replaced!!
When doing an OTA try with still Shelly bootloader there, the device is locked in last started firmware.
There is a chance left if FS is working. Go in Autoconfig and apply the config for your device.
This replaces the locked bootloader. Now you have to do the safeboot conversion, without you can not load an actual Tasmota version! The safeboot conversion will update to latest Tasmota release. For working Shelly Minis you have to update to latest Tasmota development (support is added AFTER the last release).

**Reply by ghostseven** [reply]
Hmm I don't see a safeboot conversion screen or button after setting the autoconfig and it rebooting, I assume this should be obvious?

**Reply by Th0maz** [reply]
You have to load the Partition_Wizzard.tapp file into your file system:
https://tasmota.github.io/docs/Tasmota-Application/#partition-management

**Reply by ghostseven** [reply]
Ah I am fairly sure it is indeed bricked as the .autconfig file and partition_manager.tapp are 0 bytes.

**Reply by Th0maz** [reply]
I had this before, because there's not enough free space left.
Delete the index.html.gz (or something else) from the Shelly leftovers in your file system.

**Reply by Jason2866** [reply]
The Autoconfig routine does delete the Shelly Index.html.gz file and uploads the new bootloader and PartitionWizard.
If uploading is not working (either by Autoconfig or manually) only serial flashing is the way

**Reply by ghostseven** [reply]
IT LIVES! Honestly thank you for this, absolutely fantastic assistance and apologies for my stupidity.   Removing the files that had not been cleaned up and uploading the tapp allowed me to continue the process and all is good.

**Reply by ghostseven** [reply]
Edit: Works fine! I need to do setoption21 1


Ah I spoke too soon, I think its kind of soft bricked, the power sensing does not work even though the config is right and its on 13.1.0.3 

![Screenshot 2023-09-27 at 16 07 00](https://github.com/arendst/Tasmota/assets/2887429/354608ec-cbec-4b6a-9ff4-2092f7a8a73c)

**Reply by Jason2866** [reply]
Not something that the convert process can be the reason.

**Reply by ghostseven** [reply]
It was me not doing setoption21 1 as I am a dunce (puts on big hat with D and sits in the corner)!

**Comment by xxLexanixx** [comment]
I have the same problem with my 1PM Mini
What do I have to change here so that the energy values are displayed?

![image](https://github.com/arendst/Tasmota/assets/16607717/b68249d0-7c54-47ad-b2b0-c07afde53219)

**Reply by Th0maz** [reply]
Probably update to a newer version to set GPIO7 properly.
See [this comment](https://github.com/arendst/Tasmota/discussions/19516#discussioncomment-7124608)

**Reply by xxLexanixx** [reply]
What is the correct firmware?! I don't understand which one I need

**Reply by sfromis** [reply]
Standard Tasmota build for C3 current version 13.2 should be fine.
Make sure that the template is not just stored, but also enabled.

**Comment by falcomenz** [comment]
Hi!
I have two Shelly Plus 1PM Minis. The first one got flashed successfully with mgos32-to-tasmota32 (https://github.com/tasmota/mgos32-to-tasmota32), but the second one got stuck, so i want to flash it with the serial connection. I tried flashing all kinds of different tasmota32 versions. tasmota32c3.bin, tasmota32c3-safeboot.bin and tasmota32c3.factory.bin but i always get the following error in the console:

ESP-ROM:esp32c3-api1-20210207
Build:Feb  7 2021
rst:0x10 (RTCWDT_RTC_RST),boot:0xd (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3c0a0020,len:0xf940
load:0x3fc8c600,len:0x6b0
load:0x42000020,len:0x937f8
load:0x3fc8ccb0,len:0x21a8
load:0x40380000,len:0xc514
load:0x50000000,len:0x18
SHA-256 comparison failed:
Calculated: 08fe1a6a30f32ab6a45ccadb3aa9b1cb3cfd91207eb5ed66a3ca4afa5947031d
Expected: ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
Attempting to boot anyway...
entry 0x40381d98


What di i need to flash over serial to get my Shelly working again?
Thank you :)

**Reply by sfromis** [reply]
I'm not seeing any error there. Only an informational message about SHA-256 mismatch, but this is not important as it is not encrypted firmware.

Using the [Tasmota web installer](https://tasmota.github.io/install/) is much recommended for flashing, as it automatically selects a workable binary and strategy. You told no details of what you did there. In your case, that should be `tasmota32c3.factory.bin` at offset 0, with the two others being guaranteed failures. Equally bad to not flash at the correct offset for the type of flashing.

**Reply by falcomenz** [reply]
letting the web installer do its thing and just selecting Tasmota (english) did the trick. Thank you!
